python/dgl/distributed/graph_services.py

```diff
@@ -146,7 +146,7 @@ def _sample_neighbors_graphbolt(
 
     return_eids = g.edge_attributes is not None and EID in g.edge_attributes
     subgraph = g._sample_neighbors(
-        nodes, fanout, replace=replace, return_eids=return_eids
+        nodes, None, fanout, replace=replace, return_eids=return_eids
```

Comment:
The change made here is back-compatible? Namely, if I pass `None` to `seed_offset` like here, the returned remains the same as before instead of `SampledSubgraphImpl`? As in DistGB, we don't need `SampledSubgraphImpl`. Below is how we process the returned graph from sample_neighbor():
https://github.com/Rhett-Ying/dgl/blob/fde1896f5992f4b4352eb499483be9a7fd900ea9/python/dgl/distributed/graph_services.py#L148-L162

Reply:
It is not backward conpatible right now. I currently require that the offsets are passed in the hetero case with TORCH_CHECK. I can run the new code only when offsets are passed in instead.

Reply:
That will complicate the implementation a tiny bit though.

Reply:
Have you ever tried running distDGL with CUDA backend of GraphBolt?

Reply:
For now, DistGB samples on CPU only. so this will not crash DistGB for now. But to be unified and consistent, the cpu side will apply such change as well? And we may sample on GPU someday, so I raise it up here.

Reply:
in DistGB, what we want is quite simple: given a graph, returns the sampled edges back without any post-processing. In this change, post-process is always applied.

Reply:
The CPU side is planned to have such a change. However, since the old behavior is useful, I am gonna make it optional. Does distDGL need the edges to be sorted w.r.t. edge types? How do you consume the hetero output?

Reply:
For now, DistGB doesn't care about the edges' order at all. It requires node pairs and corresponding `type_per_edge`. It can be in any order. As sampling happens on local and remote machine, when all samples finished, a new heterograph will be created then. Though it's not a quite good solution as it requires to convert from homo-to-hetero here and there, such behavior will keeps until we refactor the whole DistDGL.

Reply:
Got it, I take back some of the changes so that when seed_offsets is not passed in, old style output will be produced.

Reply:
I just tested the old behavior by manually setting seed_offsets to None, and it passes all the tests. Though I am not adding any new test to test the old behavior of the CUDA implementation in this PR.